### DISCUSSION TITLE
Supporting NextJs 13 App Directory and Server Components

### DISCUSSION THREAD

**Discussion by KlotzJesse** [iamvishnusankar/next-sitemap#574] [Ideas]
Hello,

for dynamic routes, as there is no support anymore for getServerSideProps, how would I handle dynamic routes?

- Is it possible as of today?
- if not: is there an ETA on this?
- any ideas on how this can be accomplished

I think the current best practice is to just use the still supported pages directory?

Best Regards!

**Comment by iamvishnusankar** [comment]
@KlotzJesse As of today, the solution to implement dynamic sitemap with next13 is to use the `pages` directory. I'm waiting for the stable release of `app` dir to add feature for dynamic sitemap generation, or it might be moved to the `api` route. Don't want to rush into releasing a feature at this point.

**Comment by iamvishnusankar** [comment]
@KlotzJesse Added support for appDir #591
